codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/DocumentationConverter.java

```diff
@@ -133,18 +169,56 @@ private void writeText(TextNode node) {
 
             // Docs can have valid $ characters that shouldn't run through formatters.
             String text = node.text().replace("$", "$$");
-
             if (shouldStripPrefixWhitespace) {
                 shouldStripPrefixWhitespace = false;
                 text = StringUtils.stripStart(text, " \t");
             }
 
-            if (needsListPrefix) {
-                needsListPrefix = false;
-                writer.write("");
-                writeIndent();
-                text = "* " + StringUtils.stripStart(text, " \t");
+            String curString = writer.toString();
```

Comment:
Can we operate here without having to get the entire writer body? We're re-dumping its contents on _every_ text node.

I don't see anything within the smithy writer that would let you get a subset (such as the last line) of the output, but I could be missing something.

Absent that, perhaps keep track of the previous written text, and use that instead? It looks like all of your checks are basically inspecting the end or last line of the content.